### PR TITLE
[core] Resource improvements + Async.parallelGrouped

### PR DESCRIPTION
**Pull request by fwbrasil**
- Parallel resource closing
- Allows `Abort[Throwable]` in resource close functions
- `Async.parallelGrouped`: similar to `Async.parallel` but with a max parallelism

### CODE REVIEW

File: kyo-core/shared/src/main/scala/kyo/Async.scala [hunk 1]
```diff
@@ -192,6 +192,38 @@ object Async:
...
+      * If any computation fails or is interrupted, all other computations are interrupted.
+      *
+      * @param parallelism
+      *   The maximum number of fibers to run concurrently
+      * @param seq
+      *   The sequence of computations to run in parallel
+      * @return
+      *   A sequence containing the results of all computations in their original order
+      */
+    def parallelGrouped[E, A: Flat, Ctx](parallelism: Int)(seq: Seq[A < (Abort[E] & Async & Ctx)])(
```
Comment (fwbrasil) [inline kyo-core/shared/src/main/scala/kyo/Async.scala:210]: I've added this new method to implement the parallel closing of resources.

[the remark this answers is not in the document]
Comment (fwbrasil) [reply kyo-core/shared/src/main/scala/kyo/Async.scala:210]: That was my first try. The compiler has trouble with inference in a few tests. Good point about defaults, unbounded parallelism is a common performance issue. Maybe `parallel` should always be bounded and the current unbounded APIs could become `parallelUnbounded`?

[the remark this answers is not in the document]
Comment (hearnadam) [reply kyo-core/shared/src/main/scala/kyo/Async.scala:210]: Sounds good to me.

File: kyo-combinators/shared/src/main/scala/kyo/Constructors.scala [hunk 2]
```diff
@@ -116,7 +116,7 @@ extension (kyoObject: Kyo.type)
         boundary: Boundary[Ctx, Async & Abort[E]],
         frame: Frame
     ): Seq[A1] < (Abort[E] & Async & Ctx) =
-        Async.parallel[E, A1, Ctx](sequence.map(useElement))
+        Async.parallelUnbounded[E, A1, Ctx](sequence.map(useElement))
```
Comment (hearnadam) [inline kyo-combinators/shared/src/main/scala/kyo/Constructors.scala:119]: I think we should add a bounded variant to combinators.

**Review by hearnadam — APPROVED** [review]

**Pull request merged**